File: app/androidApp/src/main/kotlin/application/liedetector/ui/screens/debug/tabs/StatesTab.kt
package application.liedetector.ui.screens.debug.tabs

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import application.liedetector.presentation.debug.DebugComponent
import application.liedetector.theme.utils.composeColor
import application.liedetector.uicore.theme.LocalDesignSystem
import application.liedetector.uicore.theme.tokens.ColorToken
import application.liedetector.uicore.theme.tokens.DimenToken
import application.liedetector.uicore.theme.tokens.StringToken
import application.liedetector.uicore.actions.DebugAction

@Composable
fun StatesTab(
    component: DebugComponent,
    contentPadding: PaddingValues = PaddingValues(0.dp)
) {
    val designSystem = LocalDesignSystem.current
    
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(contentPadding)
            .padding(designSystem.dimen(DimenToken.SPACING_MEDIUM).dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        DebugSection(title = "Engine States") {
            DebugActionButton(
                text = designSystem.string(StringToken.DEBUG_TRIGGER_LOADING),
                onClick = { component.onAction(DebugAction.TriggerLoading) },
                color = ColorToken.ACCENT_PRIMARY
            )
            DebugActionButton(
                text = designSystem.string(StringToken.DEBUG_TRIGGER_ERROR_BLOCKING),
                onClick = { component.onAction(DebugAction.TriggerErrorBlocking) },
                color = ColorToken.STRESS
            )
        }

        DebugSection(title = "Notifications / Toasts") {
            DebugActionButton(
                text = designSystem.string(StringToken.DEBUG_TRIGGER_ERROR_TOAST),
                onClick = { component.onAction(DebugAction.TriggerErrorNonBlocking) },
                color = ColorToken.ERROR
            )
            DebugActionButton(
                text = designSystem.string(StringToken.DEBUG_TRIGGER_SUCCESS_TOAST),
                onClick = { component.onAction(DebugAction.TriggerSuccessToast) },
                color = ColorToken.TRUTH
            )
        }
    }
}

@Composable
private fun DebugSection(
    title: String,
    content: @Composable ColumnScope.() -> Unit
) {
    val designSystem = LocalDesignSystem.current
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(MaterialTheme.shapes.large)
            .background(designSystem.composeColor(ColorToken.SURFACE_VARIANT).copy(alpha = 0.4f))
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        Text(
            text = title.uppercase(),
            style = MaterialTheme.typography.labelMedium,
            color = designSystem.composeColor(ColorToken.TEXT_SECONDARY),
            fontWeight = FontWeight.Bold
        )
        content()
    }
}

@Composable
private fun DebugActionButton(
    text: String,
    onClick: () -> Unit,
    color: ColorToken
) {
    val designSystem = LocalDesignSystem.current
    Button(
        onClick = onClick,
        modifier = Modifier.fillMaxWidth(),
        colors = ButtonDefaults.buttonColors(
            containerColor = designSystem.composeColor(color).copy(alpha = 0.8f),
            contentColor = designSystem.composeColor(ColorToken.TEXT_INVERTED)
        ),
        shape = MaterialTheme.shapes.medium
    ) {
        Text(text = text)
    }
}

File: app/iosApp/iosApp/UI/Debug/Tabs/StatesTab.swift
import SwiftUI
import SharedLogic

struct StatesTab: View {
    let component: DebugComponent
    let designSystem: DesignSystem
    
    var body: some View {
        ScrollView {
            VStack(spacing: 16) {
                DebugSection(title: "Engine States", designSystem: designSystem) {
                    DebugActionButton(
                        text: designSystem.string(token: .debugTriggerLoading),
                        color: .accentPrimary,
                        designSystem: designSystem,
                        action: {
                           component.onAction(action: DebugAction.TriggerLoading())
                        }
                    )
                    
                    DebugActionButton(
                        text: designSystem.string(token: .debugTriggerErrorBlocking),
                        color: .stress,
                        designSystem: designSystem,
                        action: { component.onAction(action: DebugAction.TriggerErrorBlocking()) }
                    )
                }
                
                DebugSection(title: "Notifications / Toasts", designSystem: designSystem) {
                    DebugActionButton(
                        text: designSystem.string(token: .debugTriggerErrorToast),
                        color: .error,
                        designSystem: designSystem,
                        action: { component.onAction(action: DebugAction.TriggerErrorNonBlocking()) }
                    )
                    
                    DebugActionButton(
                        text: designSystem.string(token: .debugTriggerSuccessToast),
                        color: .truth,
                        designSystem: designSystem,
                        action: { component.onAction(action: DebugAction.TriggerSuccessToast()) }
                    )
                }
            }
            .padding(CGFloat(designSystem.dimen(token: .spacingMedium)))
        }
    }
}

struct DebugSection<Content: View>: View {
    let title: String
    let designSystem: DesignSystem
    let content: Content
    
    init(title: String, designSystem: DesignSystem, @ViewBuilder content: () -> Content) {
        self.title = title
        self.designSystem = designSystem
        self.content = content()
    }
    
    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text(title.uppercased())
                .font(.caption)
                .fontWeight(.bold)
                .foregroundColor(IosTheme.color(.textSecondary, from: designSystem))
            
            VStack(spacing: 10) {
                content
            }
        }
        .padding(16)
        .background(IosTheme.color(.surfaceVariant, from: designSystem).opacity(0.4))
        .clipShape(RoundedRectangle(cornerRadius: 16, style: .continuous))
    }
}

struct DebugActionButton: View {
    let text: String
    let color: ColorToken
    let designSystem: DesignSystem
    let action: () -> Void
    
    var body: some View {
        Button(action: action) {
            Text(text)
                .font(.subheadline)
                .fontWeight(.semibold)
                .foregroundColor(IosTheme.color(.textInverted, from: designSystem))
                .frame(maxWidth: .infinity)
                .padding(.vertical, 12)
                .background(IosTheme.color(color, from: designSystem).opacity(0.8))
                .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
        }
    }
}
